tests/components/yeelight/test_config_flow.py

```diff
@@ -0,0 +1,261 @@
+"""Test the Yeelight config flow."""
+from homeassistant import config_entries
+from homeassistant.components.yeelight import (
+    CONF_DEVICE,
+    CONF_MODE_MUSIC,
+    CONF_MODEL,
+    CONF_NIGHTLIGHT_SWITCH,
+    CONF_NIGHTLIGHT_SWITCH_TYPE,
+    CONF_SAVE_ON_CHANGE,
+    CONF_TRANSITION,
+    DEFAULT_MODE_MUSIC,
+    DEFAULT_NAME,
+    DEFAULT_NIGHTLIGHT_SWITCH,
+    DEFAULT_SAVE_ON_CHANGE,
+    DEFAULT_TRANSITION,
+    DOMAIN,
+    NIGHTLIGHT_SWITCH_TYPE_LIGHT,
+)
+from homeassistant.const import CONF_ID, CONF_IP_ADDRESS, CONF_NAME
+from homeassistant.core import HomeAssistant
+
+from . import (
+    ID,
+    IP_ADDRESS,
+    MODULE,
+    MODULE_CONFIG_FLOW,
+    NAME,
+    _mocked_bulb,
+    _patch_discovery,
+)
+
+from tests.async_mock import MagicMock, patch
+from tests.common import MockConfigEntry
+
+DEFAULT_CONFIG = {
+    CONF_NAME: NAME,
+    CONF_MODEL: "",
+    CONF_TRANSITION: DEFAULT_TRANSITION,
+    CONF_MODE_MUSIC: DEFAULT_MODE_MUSIC,
+    CONF_SAVE_ON_CHANGE: DEFAULT_SAVE_ON_CHANGE,
+    CONF_NIGHTLIGHT_SWITCH: DEFAULT_NIGHTLIGHT_SWITCH,
+}
+
+
+async def test_discovery(hass: HomeAssistant):
+    """Test setting up discovery."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert not result["errors"]
+
+    with _patch_discovery(f"{MODULE_CONFIG_FLOW}.yeelight"):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {},
+        )
+    assert result2["type"] == "form"
+    assert result2["step_id"] == "pick_device"
+    assert not result2["errors"]
+
+    with patch(f"{MODULE}.async_setup", return_value=True) as mock_setup, patch(
+        f"{MODULE}.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_DEVICE: ID}
+        )
+
+    assert result3["type"] == "create_entry"
+    assert result3["title"] == NAME
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)